database/seeds/ModelSeeders/BeatmapSeeder.php

```diff
@@ -1,21 +1,32 @@
 <?php
 
+use App\Models\Beatmap;
+use App\Models\BeatmapDifficulty;
+use App\Models\BeatmapFailtimes;
+use App\Models\Beatmapset;
+use App\Models\User;
 use Carbon\Carbon;
 use Illuminate\Database\Seeder;
 
 class BeatmapSeeder extends Seeder
 {
+    // store some state to cut down querying
+    private $beatmaps = [];
+    private $beatmapsets = [];
+
     /**
      * Run the database seeds.
      *
      * @return void
      */
     public function run()
     {
-        $beatmaps_array = [];
-        $beatmapset_array = [];
-        $overbeatmaps = [];
-        $overbeatmapsets = [];
+        if ($this->command) {
```

Comment:
what's with this check? Is it not always available or something?

Reply:
`command` is optional for `Seeder`s